### PR TITLE
Improve login error behavior

### PR DESCRIPTION
**Pull request by maxmarkus**
Added loginError state handling to logout.html 
To be tested in combination with Luigi PR

### CODE REVIEW

File: core/src/logout.html [hunk 1]
```diff
@@ -62,8 +62,30 @@
...
+switch (reason) {
+  case 'tokenExpired':
+    document.getElementById('headline').innerText = 'Your session has expired.';
+    break;
+  case 'loginError':
+    document.getElementById('headline').innerText = 'There was an error while logging in.';
+    document.getElementById('message').innerText = 'Please try to log in again.';
+    break;
+  default:
+    break;
```
Comment (parostatkiem) [inline core/src/logout.html]: How about some generic error like `Ooops, something went wrong. Try to log in again`?

File: core/src/logout.html [hunk 2]
```diff
@@ -62,8 +62,30 @@
...
+  var regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
+  var results = regex.exec(location.search);
+  return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
+};
+
+var reason = getUrlParameter('reason');
+var error = getUrlParameter('error');
+
+if (error) {
+  document.getElementById('message').innerText = error;
```
Comment (parostatkiem) [inline core/src/logout.html]: In case we have error with reason 'loginError', the error won't be visible. Am I wrong?

File: core/src/logout.html [hunk 3]
```diff
@@ -62,8 +62,30 @@
   window.location.href = window.location.origin;
 }
 
-if (window.location.href.includes('tokenExpired')) {
-  document.getElementById('logout-headline').innerText = 'Your session has expired.';
+function getUrlParameter(name) {
+  name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
+  var regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
```
Comment (parostatkiem) [inline core/src/logout.html]: why `var`? It's not used outside...

File: core/src/logout.html [hunk 4]
```diff
@@ -62,8 +62,30 @@
...
+  var results = regex.exec(location.search);
+  return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
```
Comment (parostatkiem) [inline core/src/logout.html]: Why not `results ? ...`?

File: core/src/logout.html (hunk 1 above)
Comment (bszwarc) [reply core/src/logout.html]: This one sounds fine to me :)

File: core/src/logout.html (hunk 3 above)
Comment (maxmarkus) [reply core/src/logout.html]: legacy browser compatibility. it is available from ie11 on, which is the last one we support. so we could refactor to use let / const too.

File: core/src/logout.html (hunk 2 above)
Comment (maxmarkus) [reply core/src/logout.html]: Yes, this is expected.

File: core/src/logout.html (hunk 3 above)
Comment (parostatkiem) [reply core/src/logout.html]: I think I don't get it. You said IE11 supports it (which I confirm by researching it myself). So why do you still want to use `var`?

**Pull request merged**